### PR TITLE
feat: Add extra error messages when `screen` was passed instead of a DOM element

### LINKED ISSUE #911: "Please file an issue with the code that's causing you to see this error"
# Edit

**Please see my last comment – this is a case of beginner's confusion but maybe a bit of a confusing error message as well.**

- `@testing-library/dom` version: 7.30.0
- Testing Framework and version: @testing-library/react ^11.2.5
  
- DOM Environment: jest / jestdom
  

Relevant code or config

https://github.com/jdevries3133/song_maker_gallery/blob/12fcc5be423a8cdac1a439e9e35d8e6ab1fe9bc9/django_smg/frontend/src/components/teacher/teacher.test.js#L44

What you did: Working on some tests; I am still very new to testing-library and unit testing with React overall.

What happened:

I got this error message, so am opening an issue as requested:

```
Unable to find the "window" object for the given node. Please file an issue with the code that's causing you to see this error: https://github.com/testing-library/dom-testing-library/issues/new
```

Reproduction repository:
https://github.com/testing-library/dom-testing-library-template

Problem description: I got an error asking to open an issue.

Suggested solution: Who knows, but I hope this can help you trace down a bug if there is one. Let me know if you'd like any additional information.

### PR DESCRIPTION
**Pull request by savcni01**
The PR is fix #911 
When the user provides the wrong query (`screen` instead of a real query, we can output a clear, user-friendly error instead of general (and not relevant) - Create a new issue error.
**Checklist**:

- [ ] Documentation added to the "N/A"
      [docs site](https://github.com/testing-library/testing-library-docs)
- [x] Tests
- [ ] Typescript definitions updated "N/A"
- [x] Ready to be merged

### CODE REVIEW

File: src/helpers.js [hunk 1]
```diff
@@ -105,6 +105,13 @@ function getWindowFromNode(node) {
     throw new Error(
       `It looks like you passed an Array instead of a DOM node. Did you do something like \`fireEvent.click(screen.getAllBy...\` when you meant to use a \`getBy\` query \`fireEvent.click(screen.getBy...\`?`,
     )
+  } else if (
+    node.debug instanceof Function &&
```
Comment (eps1lon) [inline src/helpers.js:109]: nit:
```suggestion
    typeof node.debug === 'function' &&
```
works as well? `instanceof` always looks sketchy.

File: src/helpers.js [hunk 2]
```diff
@@ -105,6 +105,13 @@ function getWindowFromNode(node) {
     throw new Error(
       `It looks like you passed an Array instead of a DOM node. Did you do something like \`fireEvent.click(screen.getAllBy...\` when you meant to use a \`getBy\` query \`fireEvent.click(screen.getBy...\`?`,
     )
+  } else if (
+    typeof node.debug === 'function' &&
+    node.logTestingPlaygroundURL instanceof Function
```
Comment (eps1lon) [inline src/helpers.js:110]:
```suggestion
    typeof node.logTestingPlaygroundURL === 'function'
```

File: src/helpers.js [hunk 3]
```diff
@@ -105,6 +105,13 @@ function getWindowFromNode(node) {
...
+  ) {
+    throw new Error(
+      `It looks like you used 'screen' object instead of one of selectors from screen.selectMethod and it passed a 'screen' instead of a DOM node. Did you do something like \`fireEvent.click(screen, ...\` when you meant to use a \`fireEvent.click(screen.getBy..., \`?`,
```
Comment (eps1lon) [inline src/helpers.js:113]:
```suggestion
      `It looks like you passed a \`screen\` object. Did you do something like \`fireEvent.click(screen, ...\` when you meant to use a \`fireEvent.click(screen.getBy..., \`?`,
```

`screen.selectMethod` is probably too ambigious.

**Review by eps1lon — CHANGES_REQUESTED** [review]

**Comment by eps1lon** [comment]
@savcni01 Thanks!

**Comment by savcni01** [comment]
Thank you!

**Pull request merged**